src/JsonPatch.Common/JsonPatchOperationException.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+
+namespace JsonPatch
+{
+    /// <summary>
+    /// Raised when an error occurs while processing changes. Includes extra details about what was being processed.
+    /// </summary>
+    public class JsonPatchOperationException : JsonPatchException
```

Comment:
@myquay From a backwards compatibility standpoint, does it make sense to create this new exception type? Why not modify `JsonPatchException` and add a new constructor and properties to this class?

I just worry about a scenario where someone was catching this specific exception on their end during a patch operation and now it changes to `JsonPatchOperationException`.

Reply:
It should be backwards compatible. The catch block will always catch the most specific exception available to it, working its way up from the inheritance hierarchy - so a `catch(JsonPatchException ex)` will also catch the `JsonPatchOperationException`.

The main reason was to give flexibility around raising an exception in a context where it didn't make sense to return all the additional information.